{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import datetime"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If a project doesn't have these key, use empty string \"\".\n",
    "\n",
    "- title -> project_name\n",
    "- description -> project_description\n",
    "- project URL -> project_url_external\n",
    "- keywords -> sub_tag\n",
    "- category {GitHub_repo, Data, Project}\n",
    "- created_date (YYYY-MM-DD)\n",
    "- owner -> publisher\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_path = './../data/NASAOpenDataPortal/2023107_data.json'\n",
    "save_path = './../data/NASAOpenDataPortal/clean/2023107_data.json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "json_data = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Blue Origin Demo of Deorbit, Descent, and Landing Sensors (BODDL-TP)\n",
      "This contract task will mature several NASA descent and landing sensors and conduct flight tests in a relevant environment. Blue Origin, along with its NASA and JPL partners, proposes to mature critical technologies that enable precision and soft landing on the Moon. We will mature several current NASA sensors and conduct flight tests in a relevant environment. Testing will be performed at approximately 100 km altitude on-board our flight proven New Shepard vertical takeoff vertical landing (VTVL) suborbital vehicle. This proposal will allow the country to eventually meet ambitious lunar objectives, enable economic development, and support U.S. strategic leadership in space.\n",
      "https://data.nasa.gov/Aerospace/Blue-Origin-Demo-of-Deorbit-Descent-and-Landing-Se/nj3a-8wq3\n",
      "Aerospace\n",
      "Project\n",
      "2022-10-28\n",
      "\n",
      "NASA TechPort\n",
      "This dataset contains the list of all publicly published NASA projects from within the NASA technology portfolio.  In-depth portfolio information, including details for each of the NASA projects found here, can be found and searched at TechPort.NASA.gov.\n",
      "https://data.nasa.gov/Aerospace/NASA-TechPort/bq5k-hbdz\n",
      "Aerospace\n",
      "Project\n",
      "2021-12-1\n",
      "\n",
      "Testing Again Now\n",
      "A test dataset for Socrata help ticket\n",
      "https://data.nasa.gov/Space-Science/Testing-Again-Now/rg5u-t97p\n",
      "Space Science\n",
      "Project\n",
      "2021-1-19\n",
      "\n",
      "Testing Now\n",
      "A test dataset for Socrata help ticket\n",
      "https://data.nasa.gov/Space-Science/Testing-Now/3gjn-n6ht\n",
      "Space Science\n",
      "Project\n",
      "2021-1-19\n",
      "\n",
      "Testing Now\n",
      "A test dataset for Socrata help ticket\n",
      "https://data.nasa.gov/Space-Science/Testing-Now/69wk-iy5c\n",
      "Space Science\n",
      "Project\n",
      "2021-1-13\n",
      "\n",
      "NASA DAAC Locations TEST\n",
      "This is an interactive web map of NASA Distributed Active Archive Centers (DAACs). By clicking on the dots, you can learn more about what each DAAC's Area of Expertise, Location, and Website Links.\n",
      "https://data.nasa.gov/dataset/NASA-DAAC-Locations-TEST/fvg2-eqzv\n",
      "None\n",
      "Project\n",
      "2020-7-16\n",
      "\n",
      "Andersen et al 2020 SWJ Fig 1 data\n",
      "These data were used to create Fig. 1 of “Monte Carlo Evaluation of the Europa Clipper TID Margin based on the Variability of the Jovian Radiation Environment with Application for Mission Design” by Allen Andersen, Wousik Kim, Steven McClure, and Insoo Jun of the Jet Propulsion Laboratory, California Institute of Technology. \n",
      "This data set contains Galileo Energetic Particle Detector (EPD) average counts per second measurements of 3.2-10.1 MeV protons (B0), 1.5-10.5 MeV electrons (B1), and >11 MeV electrons (DC3) between 8-10 Jovian Radii (Rj) only. Each entry has a corresponding time stamp and orbit name. Further details regarding these data can be found in \n",
      "Jun, I., Garrett, H. B., Swimm, R., Evans, R. W., and Clough, G. (2005), Statistics of the variations of the high-energy electron population between 7 and 28 jovian radii as measured by the Galileo spacecraft, Icarus, 178(2), 386-394. https://doi.org/10.1016/j.icarus.2005.01.022\n",
      "The research was carried out at the Jet Propulsion Laboratory, California Institute of Technology, under a contract with the National Aeronautics and Space Administration.\n",
      "https://data.nasa.gov/dataset/Andersen-et-al-2020-SWJ-Fig-1-data/e6wj-e2uc\n",
      "None\n",
      "Project\n",
      "2019-11-25\n",
      "\n",
      "X-ray Micro-Tomography (micro-CT)\n",
      "X-ray micro-tomography (micro-CT) is a non-destructive imaging technique that provides resolution of material microstructures in three dimensions at scales from hundreds of nanometers to centimeters. The technique is used at NASA Ames Research Center to investigate a variety of advanced materials, including lightweight composite heatshields for atmospheric entry, woven materials, parachute textiles and meteoroids. Data are collected using both synchrotron and laboratory-based X-ray sources. Synchrotron micro-CT is performed in collaboration with the beamline 8.3.2 at the Advanced Light Source at Lawrence Berkeley National Laboratory. High-fidelity digital representations of microstructures obtained from micro-CT are used as framework to perform predictive simulations of effective material properties and material response using high performance computing.\n",
      "https://data.nasa.gov/Aerospace/X-ray-Micro-Tomography-micro-CT-/f7qz-8dsr\n",
      "Aerospace\n",
      "Project\n",
      "2018-10-1\n",
      "\n",
      "NASA Disasters Mapping Portal\n",
      "This entry does not contain data itself, it is for the website, the NASA Disasters Mapping Portal: https://maps.disasters.nasa.gov\n",
      "The Disasters Mapping Portal contains numerous datasets that can be streamed from the Portal into GIS software. \n",
      "The Disasters Applications area promotes the use of Earth observations to improve prediction of, preparation for, response to, and recovery from natural and technological disasters. Disaster applications and applied research on natural hazards support emergency mitigation approaches, such as early warning systems, and providing information and maps to disaster response and recovery teams.\n",
      "NOTE: Removed \"2017 - Present\" from \"Temporal Applicability\" since it's not valid\n",
      "NOTE: Removed \"Event-Specific and Near-Real Time Products\" from \"Update Frequency\" since it's not valid\n",
      "https://data.nasa.gov/Earth-Science/NASA-Disasters-Mapping-Portal/vyzu-gm3d\n",
      "Earth Science\n",
      "Project\n",
      "2018-8-6\n",
      "\n",
      "SxSW 2016 Leads\n",
      "Companies that visited the NASA boot at the SxSW 2016 conference.\n",
      "https://data.nasa.gov/Management-Operations/SxSW-2016-Leads/yvxp-ccvk\n",
      "Management/Operations\n",
      "Project\n",
      "2016-3-24\n",
      "\n",
      "Global Landslide Catalog Export\n",
      "The Global Landslide Catalog (GLC) was developed with the goal of identifying rainfall-triggered landslide events around the world, regardless of size, impacts or location. The GLC considers all types of mass movements triggered by rainfall, which have been reported in the media, disaster databases, scientific reports, or other sources. The GLC has been compiled since 2007 at NASA Goddard Space Flight Center. This is a unique data set with the ID tag “GLC” in the landslide editor.\n",
      "This dataset on data.nasa.gov was a one-time export from the Global Landslide Catalog maintained separately. It is current as of March 7, 2016. The original catalog is available here: http://www.arcgis.com/home/webmap/viewer.html?url=https%3A%2F%2Fmaps.nccs.nasa.gov%2Fserver%2Frest%2Fservices%2Fglobal_landslide_catalog%2Fglc_viewer_service%2FFeatureServer&source=sd\n",
      "To export GLC data, you must agree to the “Terms and Conditions”. We request that anyone using the GLC cite the two sources of this database:\n",
      "Kirschbaum, D. B., Adler, R., Hong, Y., Hill, S., & Lerner-Lam, A. (2010). A global landslide catalog for hazard applications: method, results, and limitations. Natural Hazards, 52(3), 561–575. doi:10.1007/s11069-009-9401-4. [1]\n",
      "Kirschbaum, D.B., T. Stanley, Y. Zhou (In press, 2015). Spatial and Temporal Analysis of a Global Landslide Catalog. Geomorphology. doi:10.1016/j.geomorph.2015.03.016. [2]\n",
      "https://data.nasa.gov/Earth-Science/Global-Landslide-Catalog-Export/dd9e-wu2v\n",
      "Earth Science\n",
      "Project\n",
      "2016-3-7\n",
      "\n",
      "Outgassing Db\n",
      "This tenth compilation of outgassing data of materials intended for spacecraft use supersedes Reference Publication 1124, Revision 3, September 1993. The data were obtained at the Goddard Space Flight Center (GSFC), utilizing equipment developed at Stanford Research Institute (SRI) under contract to the Jet Propulsion Laboratory (JPL).\n",
      "https://data.nasa.gov/Applied-Science/Outgassing-Db/r588-f7pr\n",
      "Applied Science\n",
      "Project\n",
      "2015-11-19\n",
      "\n",
      "Candida albicans response to spaceflight (NASA STS-115) --- GSM1231690_Slide_43\n",
      "https://c3.nasa.gov/genelab/accession/GLDS-20/\n",
      "This study presents the first global transcriptional profiling and phenotypic characterization of the major human opportunistic fungal pathogen, Candida albicans, grown in spaceflight conditions. Microarray analysis revealed that C. albicans subjected to short-term spaceflight culture differentially regulated 454 genes compared to synchronous ground controls, which represented 8.4% of the analyzed ORFs. Spaceflight-cultured C. albicans induced genes involved in cell aggregation (similar to flocculation), which was validated by microscopic and flow cytometry analysis. We also observed enhanced random budding of spaceflight-cultured cells as opposed to more normal bipolar budding patterns for ground samples, in accordance with the gene expression data. Furthermore, genes involved in antifungal agent and stress resistance were differentially regulated in spaceflight, including induction of ABC transporters and members of the major facilitator family, downregulation of ergosterol-encoding genes, and upregulation of genes involved in oxidative stress resistance. Finally, downregulation of genes involved in the actin cytoskeleton was observed. Interestingly, the transcriptional regulator Cap1 and over 30% of the Cap1 regulon was differentially expressed in spaceflight-cultured C. albicans. A potential role for Cap1 in the spaceflight response of C. albicans is suggested, as this regulator is involved in random budding, cell aggregation, actin cytoskeleton, and oxidative stress resistance; all related to observed spaceflight-associated changes of C. albicans. While culture of C. albicans in microgravity potentiates a global change in gene expression that could induce a virulence-related phenotype, no increased virulence in a murine intraperitoneal (i.p.) infection model was observed. This study represents an important basis for the assessment of the risk that commensal flora could play during spaceflight missions. Furthermore, since the low fluid-shear environment of microgravity is relevant to physical forces encountered by pathogens during the infection process, insights gained from this study could identify novel infectious disease mechanisms, with downstream benefits for the general public. Cells were grown for 24 hours on the space shuttle or as ground-based controls, preserved in RNALater, and stored at -80C. Four samples of each flight- and ground-based controls were harvested for microarray analysis. GAP is Group Activation Pack and each GAP contains 8 FPAs. The numbers represent the # assigned to the particular GAP and the number assigned to the specific FPA (1-8) within the indicated GAP. The same hardware is used for the flight samples and the ground samples.\n",
      "https://data.nasa.gov/Applied-Science/Candida-albicans-response-to-spaceflight-NASA-STS-/59ui-jv2j\n",
      "Applied Science\n",
      "Project\n",
      "2015-10-14\n",
      "\n",
      "S E- GEOD-50881 Study Samples --- Candida albicans response to spaceflight (NASA STS-115)\n",
      "https://c3.nasa.gov/genelab/accession/GLDS-20/\n",
      "This study presents the first global transcriptional profiling and phenotypic characterization of the major human opportunistic fungal pathogen, Candida albicans, grown in spaceflight conditions. Microarray analysis revealed that C. albicans subjected to short-term spaceflight culture differentially regulated 454 genes compared to synchronous ground controls, which represented 8.4% of the analyzed ORFs. Spaceflight-cultured C. albicans induced genes involved in cell aggregation (similar to flocculation), which was validated by microscopic and flow cytometry analysis. We also observed enhanced random budding of spaceflight-cultured cells as opposed to more normal bipolar budding patterns for ground samples, in accordance with the gene expression data. Furthermore, genes involved in antifungal agent and stress resistance were differentially regulated in spaceflight, including induction of ABC transporters and members of the major facilitator family, downregulation of ergosterol-encoding genes, and upregulation of genes involved in oxidative stress resistance. Finally, downregulation of genes involved in the actin cytoskeleton was observed. Interestingly, the transcriptional regulator Cap1 and over 30% of the Cap1 regulon was differentially expressed in spaceflight-cultured C. albicans. A potential role for Cap1 in the spaceflight response of C. albicans is suggested, as this regulator is involved in random budding, cell aggregation, actin cytoskeleton, and oxidative stress resistance; all related to observed spaceflight-associated changes of C. albicans. While culture of C. albicans in microgravity potentiates a global change in gene expression that could induce a virulence-related phenotype, no increased virulence in a murine intraperitoneal (i.p.) infection model was observed. This study represents an important basis for the assessment of the risk that commensal flora could play during spaceflight missions. Furthermore, since the low fluid-shear environment of microgravity is relevant to physical forces encountered by pathogens during the infection process, insights gained from this study could identify novel infectious disease mechanisms, with downstream benefits for the general public. Cells were grown for 24 hours on the space shuttle or as ground-based controls, preserved in RNALater, and stored at -80C. Four samples of each flight- and ground-based controls were harvested for microarray analysis. GAP is Group Activation Pack and each GAP contains 8 FPAs. The numbers represent the # assigned to the particular GAP and the number assigned to the specific FPA (1-8) within the indicated GAP. The same hardware is used for the flight samples and the ground samples.\n",
      "https://data.nasa.gov/Applied-Science/S-E-GEOD-50881-Study-Samples-Candida-albicans-resp/8e7s-kdza\n",
      "Applied Science\n",
      "Project\n",
      "2015-10-14\n",
      "\n",
      "A E- GEOD-50881 Gene Chip Assay --- Candida albicans response to spaceflight (NASA STS-115)\n",
      "https://c3.nasa.gov/genelab/accession/GLDS-20/\n",
      "This study presents the first global transcriptional profiling and phenotypic characterization of the major human opportunistic fungal pathogen, Candida albicans, grown in spaceflight conditions. Microarray analysis revealed that C. albicans subjected to short-term spaceflight culture differentially regulated 454 genes compared to synchronous ground controls, which represented 8.4% of the analyzed ORFs. Spaceflight-cultured C. albicans induced genes involved in cell aggregation (similar to flocculation), which was validated by microscopic and flow cytometry analysis. We also observed enhanced random budding of spaceflight-cultured cells as opposed to more normal bipolar budding patterns for ground samples, in accordance with the gene expression data. Furthermore, genes involved in antifungal agent and stress resistance were differentially regulated in spaceflight, including induction of ABC transporters and members of the major facilitator family, downregulation of ergosterol-encoding genes, and upregulation of genes involved in oxidative stress resistance. Finally, downregulation of genes involved in the actin cytoskeleton was observed. Interestingly, the transcriptional regulator Cap1 and over 30% of the Cap1 regulon was differentially expressed in spaceflight-cultured C. albicans. A potential role for Cap1 in the spaceflight response of C. albicans is suggested, as this regulator is involved in random budding, cell aggregation, actin cytoskeleton, and oxidative stress resistance; all related to observed spaceflight-associated changes of C. albicans. While culture of C. albicans in microgravity potentiates a global change in gene expression that could induce a virulence-related phenotype, no increased virulence in a murine intraperitoneal (i.p.) infection model was observed. This study represents an important basis for the assessment of the risk that commensal flora could play during spaceflight missions. Furthermore, since the low fluid-shear environment of microgravity is relevant to physical forces encountered by pathogens during the infection process, insights gained from this study could identify novel infectious disease mechanisms, with downstream benefits for the general public. Cells were grown for 24 hours on the space shuttle or as ground-based controls, preserved in RNALater, and stored at -80C. Four samples of each flight- and ground-based controls were harvested for microarray analysis. GAP is Group Activation Pack and each GAP contains 8 FPAs. The numbers represent the # assigned to the particular GAP and the number assigned to the specific FPA (1-8) within the indicated GAP. The same hardware is used for the flight samples and the ground samples.\n",
      "https://data.nasa.gov/Applied-Science/A-E-GEOD-50881-Gene-Chip-Assay-Candida-albicans-re/c5py-4h4g\n",
      "Applied Science\n",
      "Project\n",
      "2015-10-14\n",
      "\n",
      "NASA Facilities\n",
      "None\n",
      "https://data.nasa.gov/Management-Operations/NASA-Facilities/gvk9-iz74\n",
      "Management/Operations\n",
      "Project\n",
      "2015-8-26\n",
      "\n",
      "BeXRB Monitor Data\n",
      "This is a monitor using data from Swift/BAT, MAXI and Fermi/GBM instruments. \n",
      "Click on the table header to change the sorting behaviour. \n",
      "Color code for the probability: (rising flux , no change, decreasing flux)\n",
      "The average flux is calculated as the average of the last 3 days of data, without taking the errors into account. The fermi flux is not given in mCrab. \n",
      "Please note that the accuracy of the probabilities is limited by the accuracy of the data used.\n",
      "Release of Friday, the 17th of October 2014\n",
      "Last updated at 14:04:07 UT\n",
      "Created as an ESAC trainee project by Eva Laplace. \n",
      "Supervisors: Peter Kretschmar, Emilio Salazar and Miguel Arregui. \n",
      "Contact: peter.kretschmar@esa.int\n",
      "https://data.nasa.gov/dataset/BeXRB-Monitor-Data/jdkf-j3pt\n",
      "None\n",
      "Project\n",
      "2015-4-12\n",
      "\n",
      "Near-Earth Comets - Orbital Elements\n",
      "The table below provides J2000 heliocentric ecliptic orbital elements\n",
      "for 170 NECs (Near-Earth Comets) sorted by object number/name.\n",
      "http://neo.jpl.nasa.gov/cgi-bin/neo_elem?type=NEC\n",
      "https://data.nasa.gov/Space-Science/Near-Earth-Comets-Orbital-Elements/b67r-rgxc\n",
      "Space Science\n",
      "Project\n",
      "2015-4-2\n",
      "\n",
      "Meteorite Landings\n",
      "This comprehensive data set from The Meteoritical Society contains information on all of the known meteorite landings. The Fusion Table is collected by Javier de la Torre and we've also provided an XLS file that consists of 34,513 meteorites and includes the following fields:\n",
      "place\n",
      "type_of_meteorite\n",
      "mass_g\n",
      "fell_found\n",
      "year\n",
      "database\n",
      "coordinate_1\n",
      "coordinates_2\n",
      "cartodb_id\n",
      "created_at\n",
      "updated_at\n",
      "year_date\n",
      "longitude\n",
      "latitude\n",
      "geojson\n",
      "**5/14/13\n",
      "Please find an updated data set from The Meteoritical Society that includes more recent meteorites. Under NameType, 'valid' is for most meteorites and 'relict' are for objects that were once meteorites but are now highly altered by weathering on Earth.\n",
      "http://visualizing.org/datasets/meteorite-landings\n",
      "https://data.nasa.gov/Space-Science/Meteorite-Landings/gh4g-9sfh\n",
      "Space Science\n",
      "Project\n",
      "2015-4-2\n",
      "\n",
      "Fireball And Bolide Reports\n",
      "Fireballs and bolides are astronomical terms for exceptionally bright meteors that are spectacular enough to to be seen over a very wide area. The following table provides a chronological data summary of fireball and bolide events provided by U.S. Government sensors. Ground-based observers sometimes also witness these events at night, or much more rarely in daylight, as impressive atmospheric light displays. This website is not meant to be a complete list of all fireball events. Only the brightest fireballs are noted.\n",
      "A meteoroid is generally defined as an asteroid or comet fragment that orbits the Sun and has an approximate size between ten microns and a meter or so. Meteors, or \"shooting stars,\" are the visible paths of meteoroids that have entered the Earth's atmosphere at high velocities. A fireball is an unusually bright meteor that reaches a visual magnitude of -3 or brighter when seen at the observer's zenith. Objects causing fireball events can exceed one meter in size. Fireballs that explode in the atmosphere are technically referred to as bolides although the terms fireballs and bolides are often used interchangeably.\n",
      "During the atmospheric entry phase, an impacting object is both slowed and heated by atmospheric friction. In front of it, a bow shock develops where atmospheric gases are compressed and heated. Some of this energy is radiated to the object causing it to ablate, and in most cases, to break apart. Fragmentation increases the amount of atmosphere intercepted and so enhances ablation and atmospheric braking. The object catastrophically disrupts when the force from the unequal pressures on the front and back sides exceeds its tensile strength.\n",
      "Objects causing fireballs are usually not large enough to survive passage through the Earth's atmosphere intact, although fragments, or meteorites, are sometimes recovered on the ground. The approximate total radiated energy in the atmosphere is provided in unit of Joules, a unit of energy given in kilograms times velocity squared, or kg x (m/s)2. An event with an energy equivalent of one thousand tons of TNT explosives is termed a kiloton (kt) event, where 1 kt = 4.185 x 1012 Joules. In the accompanying table, the total radiated energy is given but this is always less that the total impact energy. Peter Brown and colleagues have provided an empirical expression to approximately provide the total impact energy in kt (E), given the optical radiant energy in kt (Eo) (see: Brown et al., The flux of small near-Earth objects colliding with the Earth. Nature, vol. 420, 21 Nov. 2002, pp. 294-296).\n",
      "E = 8.2508 x Eo0.885\n",
      "The accompanying table provides information on the date and time of each fireball event, its geographic location, its altitude and velocity at peak brightness, its approximate total optical radiated energy and its calculated total impact energy. The pre-impact velocity components are expressed in a geocentric Earth-fixed reference frame defined as follows: the z-axis is directed along the Earth's rotation axis towards the celestial north pole, the x-axis lies in the Earth's equatorial plane, directed towards the prime meridian, and the y-axis completes the right-handed coordinate system.\n",
      "http://neo.jpl.nasa.gov/fireball/\n",
      "https://data.nasa.gov/Space-Science/Fireball-And-Bolide-Reports/mc52-syum\n",
      "Space Science\n",
      "Project\n",
      "2015-4-2\n",
      "\n",
      "WISE NEA/COMET DISCOVERY STATISTICS\n",
      "These tables show discovery statistics for NEAs and comets discovered by NASA's WISE mission - now renamed to NEOWISE. The first small table shows the number of NEAs, PHAs (a sub-group of NEAs), and comets discovered to-date (within a day or two). The second table shows each object discovered, sorted by designation, with selected parameters describing the object's orbit.\n",
      "http://neo.jpl.nasa.gov/stats/wise/\n",
      "https://data.nasa.gov/Space-Science/WISE-NEA-COMET-DISCOVERY-STATISTICS/7qz6-zrqt\n",
      "Space Science\n",
      "Project\n",
      "2015-4-2\n",
      "\n",
      "Extra-vehicular Activity (EVA) - US and Russia\n",
      "Activities done by an astronaut or cosmonaut outside a spacecraft beyond the Earth's appreciable atmosphere.\n",
      "https://data.nasa.gov/Raw-Data/Extra-vehicular-Activity-EVA-US-and-Russia/9kcy-zwvn\n",
      "Raw Data\n",
      "Project\n",
      "2014-7-31\n",
      "\n",
      "NASA Patents\n",
      "This dataset shows information pertaining to NASA held and pending patents.\n",
      "https://data.nasa.gov/Raw-Data/NASA-Patents/gquh-watm\n",
      "Raw Data\n",
      "Project\n",
      "2014-7-30\n",
      "\n",
      "Open Source And General Resource Software\n",
      "This dataset lists out all software in use by NASA\n",
      "https://data.nasa.gov/Software/Open-Source-And-General-Resource-Software/fk38-4khf\n",
      "Software\n",
      "Project\n",
      "2014-7-30\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Open the JSON file for reading\n",
    "with open(file_path, 'r') as file:\n",
    "    # Load the JSON data\n",
    "    data = json.load(file)\n",
    "    json_len = len(data)\n",
    "\n",
    "    for index in range(json_len):\n",
    "        project_name = data[index]['title']\n",
    "        project_description = data[index]['description']\n",
    "        project_url_external = data[index]['Project URL']\n",
    "        sub_tag =  data[index]['keywords']\n",
    "        category = 'Project'\n",
    "        publisher = \"\"\n",
    "\n",
    "        # formatting the date, January 19 2021\n",
    "        dateString = data[index]['date']\n",
    "        dateFormatter = \"%B %d %Y\"\n",
    "        datetime_object = datetime.datetime.strptime(dateString, dateFormatter)\n",
    "        created_date = str(datetime_object.year) + '-' + str(datetime_object.month) + '-' + str(datetime_object.day)\n",
    "        \n",
    "        # print(index)\n",
    "        print(project_name)\n",
    "        print(project_description)\n",
    "        print(project_url_external)\n",
    "        print(sub_tag)\n",
    "        print(category)\n",
    "        print(created_date)\n",
    "        print(publisher)\n",
    "\n",
    "        json_data.append({\"project_name\": project_name, \"project_description\": project_description, \"project_url_external\": project_url_external, \"sub_tag\": sub_tag, \"category\": category, \"created_date\": created_date, \"publisher\": publisher})\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "json_data = json.dumps(json_data, indent=4)  # 'indent' for pretty formatting\n",
    "\n",
    "# Write the JSON string to the file\n",
    "with open(save_path, 'w', encoding='utf-8') as json_file:\n",
    "    json_file.write(json_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "WebCrawler",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
